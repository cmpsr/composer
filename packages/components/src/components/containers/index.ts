export * from './Box';
export * from './Grid';
export * from './Section';
export * from './SectionChild';
