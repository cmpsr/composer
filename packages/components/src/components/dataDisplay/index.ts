export { ListItem } from '@chakra-ui/react';
export * from './Badge';
export * from './Divider';
export * from './NumberedList';
export * from './Symbol';
export * from './Tag';
export * from './UnorderedList';
