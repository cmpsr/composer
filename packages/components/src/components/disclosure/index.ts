export * from './Accordion';
export * from './Tabs';
