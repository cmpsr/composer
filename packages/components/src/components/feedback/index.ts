export * from './Alert';
export * from './InlineAlert';
export * from './ProgressBar';
export * from './ProgressCircular';
export * from './Skeleton';
export * from './Spinner';
export * from './Toast';
