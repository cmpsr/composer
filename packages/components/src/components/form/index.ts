export * from './Autocomplete';
export * from './AutocompleteMultiSelect';
export * from './Button';
export * from './ButtonIcon';
export * from './Checkbox';
export * from './DateInput';
export * from './IconButton';
export * from './Input';
export * from './MaskInput';
export * from './NumberInput';
export * from './PhoneInput';
export * from './PinInput';
export * from './Radio';
export * from './RangeSlider';
export * from './SegmentedButton';
export * from './Select';
export * from './Slider';
export * from './Switch';
export * from './Textarea';
