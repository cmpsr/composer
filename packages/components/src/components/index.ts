export * from './layouts';
export * from './primitives';
export * from './navigation';
export * from './notifications';
