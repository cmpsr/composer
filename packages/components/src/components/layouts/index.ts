export * from './Box';
export * from './Flex';
export * from './Grid';
export * from './Section';
