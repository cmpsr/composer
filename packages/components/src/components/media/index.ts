export * from './Avatar';
export * from './Image';
export * from './Icons';
