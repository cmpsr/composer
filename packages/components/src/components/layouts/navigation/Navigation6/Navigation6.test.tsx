import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Navigation6, NAVIGATION6_DEFAULT_TEST_ID } from '.';
import { BOX_DEFAULT_TEST_ID } from 'components/containers';

describe('Navigation6', () => {
  const testStyle = { background: { default: { bg: 'white' } } };
  it('should match snapshot', () => {
    const navigation = renderer.create(<Navigation6 style={testStyle} />);
    const tree = navigation.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render component', () => {
    const { getByTestId } = render(<Navigation6 style={testStyle} />);
    const navigation = getByTestId(NAVIGATION6_DEFAULT_TEST_ID);
    expect(navigation).toBeInTheDocument();
  });
  it('should render style', () => {
    const { getByTestId } = render(<Navigation6 style={testStyle} />);
    const navigation = getByTestId(BOX_DEFAULT_TEST_ID);
    expect(navigation.classList).toContain('bg-white');
  });
  it('should render content logo', () => {
    const childrenLogoTestId = 'Children-Logo-TestId';
    const renderedChildren = {
      logo: <p data-testid={childrenLogoTestId}>Logo</p>,
      content: <div>Children</div>,
    };
    const { getByTestId } = render(
      <Navigation6 style={testStyle}>{renderedChildren}</Navigation6>,
    );
    const childrenLogo = getByTestId(childrenLogoTestId);
    expect(childrenLogo).toBeInTheDocument();
    expect(childrenLogo.parentElement.classList).toContain('flex-none');
  });
  it('should render array of content with appropriate style', () => {
    const childrenTestId = 'Children-TestId';
    const renderedChildren = {
      logo: <p>Logo</p>,
      content: [
        <div data-testid={childrenTestId + '1'} key="1">
          Children1
        </div>,
        <div data-testid={childrenTestId + '2'} key="2">
          Children2
        </div>,
      ],
    };
    const { baseElement, getAllByTestId } = render(
      <Navigation6 style={testStyle}>{renderedChildren}</Navigation6>,
    );

    const children = getAllByTestId(childrenTestId, { exact: false });
    expect(children.length).toBe(2);
    expect(baseElement.innerHTML).toContain('justify-start');
  });
  it('should render buttons', () => {
    const childrenButtonsId = 'Children-ButtonsId';
    const renderedChildren = {
      logo: <p>Logo</p>,
      secondaryContent: [
        <button data-testid={childrenButtonsId + '1'} key="1">
          Sign in
        </button>,
        <button data-testid={childrenButtonsId + '2'} key="2">
          Log in
        </button>,
      ],
    };

    const { baseElement, getAllByTestId } = render(
      <Navigation6 style={testStyle}>{renderedChildren}</Navigation6>,
    );
    const children = getAllByTestId(childrenButtonsId, { exact: false });
    expect(children.length).toBe(2);
    expect(baseElement.innerHTML).toContain('grid-cols-2');
  });
});
