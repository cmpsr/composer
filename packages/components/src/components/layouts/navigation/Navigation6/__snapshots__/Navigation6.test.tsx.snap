// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation6 should match snapshot 1`] = `
<div
  className="w-full h-full py-8 bg-white"
  data-testid="box"
>
  <nav
    className="w-full h-full max-w-content mx-auto items-center flex flex-col md:flex-row"
    data-testid="navigation6"
  >
    <div
      className="flex-none"
    />
    <div
      className="flex flex-col md:flex-row flex-1 w-full justify-start ml-10"
    >
      <div
        className="flex flex-grow-0 text-center my-4 md:my-0"
      />
    </div>
    <div
      className="flex-1 flex justify-end"
    />
  </nav>
</div>
`;
